src/components/EditAddSensorModal/index.tsx

```diff
@@ -0,0 +1,380 @@
+                errors={formatError(errors.name?.message)}
+                containerClassName='xs:col-span-9'
+              />
+            )}
+          />
+          <Controller
+            name='symbolId'
+            control={control}
+            defaultValue={defaultValues?.symbolId}
+            render={({ field }) => (
+              <FormListBox
+                {...field}
+                label='Symbol'
+                containsIconList
+                placeholder='Symbol'
+                options={Array.from(Array(32)).map((_, i) => ({
+                  name: (
+                    <span className='w-6 h-6 float-left inline-block'>
+                      <SensorSymbol symbol={i + 1} />
+                    </span>
+                  ),
+                  value: i + 1,
+                }))}
+                errors={formatError(errors.symbolId?.message)}
+                className='xs:col-span-3'
+              />
+            )}
+          />
+        </fieldset>
+        <Controller
+          name='description'
+          control={control}
+          defaultValue={defaultValues?.description}
+          render={({ field }) => (
+            <FormTextarea
+              {...field}
+              label='Beschreibung'
+              placeholder='Beschreibe kurz der Sensor'
+              errors={formatError(errors.description?.message)}
+            />
+          )}
+        />
+        <fieldset className='sm:grid sm:grid-cols-2 gap-4 relative z-50'>
+          <Controller
+            name='categoryId'
+            control={control}
+            defaultValue={defaultValues?.categoryId}
+            render={({ field }) => (
+              <FormListBox
+                {...field}
+                label='Kategorie'
+                placeholder='Wähle eine Kategorie'
+                options={
+                  (categories &&
+                    !isLoadingCategories &&
+                    !categoriesError &&
+                    categories.map(category => ({
+                      value: category.id,
+                      name: (
+                        <span className='flex items-center gap-2'>
+                          <CategoryIcon categoryId={category.id} />
+                          {category.name}
+                        </span>
+                      ),
+                    }))) ||
+                  []
+                }
+                errors={formatError(errors.categoryId?.message)}
+              />
+            )}
+          />
+          <div className='flex flex-col gap-2'>
+            <Controller
+              name='integration'
+              control={control}
+              defaultValue={defaultValues?.integration || integration}
+              render={({ field }) => (
+                <FormListBox
+                  {...field}
+                  onChange={newValue => {
+                    field.onChange(newValue);
+                    setIntegration(newValue as "http" | "ttn");
```

Comment:
Isn't it possible to set this to `as definitions["sensors"]["connection_type"]`? This way we don't have to manually update this once the `connection_type`s change

Reply:
The problem is that the backend currently also accepts "other" but the frontend doesn't. I will omit "other".

Reply:
I tried omit but it isn't possible on enums... :(